projects/OG-Engine/src/main/java/com/opengamma/engine/marketdata/spec/MarketDataSpecificationParser.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.engine.marketdata.spec;
+
+import org.threeten.bp.LocalDate;
+import org.threeten.bp.format.DateTimeParseException;
+
+import com.opengamma.id.UniqueId;
+
+/**
+ * Parses a string to produce a {@link MarketDataSpecification}.
+ */
+public class MarketDataSpecificationParser {
```

Comment:
Should have a private constructor as there is no state and nothing but static methods.